integrations/huggingface-transformers/tutorials/sentiment-analysis/sentiment-analysis-cli.md

```diff
@@ -0,0 +1,328 @@
+
+For simplicity, we will perform the fine-tuning without distillation. Although the transfer learning recipe contains distillation
+modifiers, by setting `--distill_teacher disable` we instruct SparseML to skip distillation.
+
+### Rotten Tomatoes Inspection
+
+Run the following to inspect the Rotten Tomatoes dataset.
+
+```python
+from datasets import load_dataset
+
+rotten_tomatoes = load_dataset("rotten_tomatoes")
+print(rotten_tomatoes)
+print(rotten_tomatoes["train"][0])
+```
+
+We can see that each row dataset contains a `text` field which is a string representing the sequence to be classified and a `label` field which is a `0` or `1` representing negative and positive labels.
+
+### Using a Hugging Face Dataset Identifier
+
+We can easily pass the `rotten_tomatoes` Hugging Face dataset identifier to train with this dataset instead of SST2. Simply replace the `--task_name sst2` argument with `--dataset_name rotten_tomatoes --input_column_names text --label_column_name label`. SparseML will then download the dataset from the Hugging Face hub and run training as before.
+
+```
+sparseml.transformers.text_classification \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --recipe zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none \
+  --recipe_args '{"num_epochs":12,"qat_start_epoch":7.0, "observer_epoch": 11.0}' \
+  --distill_teacher disable \
+  --dataset_name rotten_tomatoes --input_column_names "text" --label_column_name "label" \
+  --output_dir sparse_quantized_bert-text_classification_rotten_tomatoes-hf_dataset --max_seq_length 128 --per_device_train_batch_size 32 \
+  --per_device_eval_batch_size 32 --preprocessing_num_workers 6 --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+You will notice that we used the same recipe as we did in the SST2 case (identified by the SparseZoo stub `zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none`). Since the Rotten Tomatoes and SST2 tasks are similiar, we chose to start with the same hyperparameters as we used in SST2 training. 
+
+#### Recipe Args
+
+To update a recipe and experiment with hyperparameters, you can download the YAML file from SparseZoo, make updates to the YAML directly, and pass the local path to SparseML.
+
+Alternatively, you can use `--recipe_args` to modify the recipe on the fly. In this case, we used the following to run for 12 epochs instead of 13:
+
+```bash
+--recipe_args '{"num_epochs":12,"qat_start_epoch":7.0, "observer_epoch": 11.0}'
+```
+
+### Using Local CSV/JSON Files
+
+Let's walk through how to pass a CSV/JSON dataset to the CLI.
+
+#### Save Dataset as a CSV File
+
+For this example, we use Hugging Face `datasets` to create a CSV file for Rotten Tomatoes that can be passed to SparseML's CLI but you can use any framework you want to create the CSV.
```

Comment:
For this example, we use Hugging Face `datasets` to create a CSV file for Rotten Tomatoes that can be passed to SparseML's CLI, but you can use any framework you want to create the CSV.